src/WebJobs.Script/Binding/Http/HttpExtensions.cs

```diff
@@ -0,0 +1,284 @@
+            JToken jToken = null;
+
+            try
+            {
+                jToken = JToken.Parse(value);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("Invalid Json object! " + ex.Message);
+            }
+            request.Deserialize(jToken);
+        }
+
+        public static void Deserialize(this HttpRequestMessage request, JToken jsonToken)
+        {
+            if (jsonToken["Method"] != null)
+            {
+                request.Method = JsonConvert.DeserializeObject<HttpMethod>(jsonToken["Method"].ToString());
+            }
+
+            if (jsonToken["RequestUri"] != null)
+            {
+                Uri uri = null;
+                if (Uri.TryCreate(jsonToken["RequestUri"].ToString(), UriKind.Absolute, out uri))
+                {
+                    request.RequestUri = uri;
+                }
+            }
+
+            var content = jsonToken["Content"];
+            if (content != null && content.HasValues)
+            {
+                DeserializeContent(request, content);
+            }
+
+            if (jsonToken["Headers"] != null && jsonToken["Headers"].HasValues)
+            {
+                foreach (var header in jsonToken["Headers"].Children())
+                {
+                    AddHeader(request.Headers, header);
+                }
+            }
+        }
+
+        public static void Deserialize(this HttpResponseMessage res, string response)
+        {
+            JObject jObject = null;
+
+            try
+            {
+                jObject = JObject.Parse(response);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("Invalid Json object! " + ex.Message);
+            }
+
+            HttpStatusCode statusCode;
+            if (jObject["StatusCode"] != null && Enum.TryParse(jObject["StatusCode"].ToString(), out statusCode))
+            {
+                res.StatusCode = statusCode;
+            }
+
+            if (jObject["ReasonPhrase"] != null)
+            {
+                res.ReasonPhrase = jObject["ReasonPhrase"].ToString();
+            }
+
+            var content = jObject["Content"];
+            if (content != null && content.HasValues)
+            {
+                DeserializeContent(res, content);
+            }
+
+            if (jObject["Headers"] != null && jObject["Headers"].HasValues)
+            {
+                foreach (var header in jObject["Headers"].Children())
+                {
+                    AddHeader(res.Headers, header);
+                }
+            }
+
+            if (jObject["RequestMessage"] != null && jObject["RequestMessage"].HasValues)
+            {
+                res.RequestMessage = new HttpRequestMessage();
+                res.RequestMessage.Deserialize(jObject["RequestMessage"]);
+            }
+        }
+
+        private static void DeserializeContent(dynamic req, JToken content)
```

Comment:
Once we finalize all these helpers, we'll need good unit tests for all this stuff.